### PR TITLE
Yet another box period search

### PR DESCRIPTION
**Pull request by mirca**
``fast_box_period_search`` is 2x faster than ``box_period_search`` and it seems surprisingly more
accurate (although I didn't do many experiments to support this).

screen shot 2018-01-26 at 11 52 37 am
screen shot 2018-01-26 at 11 52 49 am

my notes on bls:
[bls.pdf](https://github.com/KeplerGO/lightkurve/files/1669172/bls.pdf)

todo: add docstrings

### CODE REVIEW

**Comment by barentsen** [comment]
wowwwww

**Comment by barentsen** [comment]
Impressive!!  Two questions:
- In the numerical optimization, do you know how many iterations there are?  I ask to understand how much slower or faster it would be to do a grid search.
- The advantage of a grid search may be that you'd be able to deal with a multi-modal case of two planets in the same period, but I guess planets like that don't even exist.

**Comment by barentsen** [comment]
Your PDF is so much easier to understand and follow than the BLS paper and its funny weights.

**Comment by mirca** [comment]
> In the numerical optimization, do you know how many iterations there are? I ask to understand how much slower or faster it would be to do a grid search.

It takes on average 3 iterations

**Comment by barentsen** [comment]
Do you mean there are 3 iterations between the numerical and the analytical step, or the numerical optimization only requires 3 steps to find the maximum likelihood?  I'm wondering about the latter.

**Comment by mirca** [comment]
> Do you mean there are 3 iterations between the numerical and the analytical step, or the numerical optimization only requires 3 steps to find the maximum likelihood? I'm wondering about the latter.

The latter.

**Comment by mirca** [comment]
The marginalized loglikehood as a function of ``to`` and ``w`` respectively
screen shot 2018-01-26 at 2 48 13 pm
screen shot 2018-01-26 at 2 48 20 pm

**Comment by mirca** [comment]
@barentsen I plan to merge this provided tests pass. How does that sound?

File: lightkurve/lightcurve.py [hunk 1]
```diff
@@ -1224,3 +1226,114 @@ def box(amplitude, depth, to, width):
...
+    log_posterior = []
+    for p in tqdm(trial_periods):
+        folded = lc.fold(period=p)
+        amplitude_star, depth_star = m, 1e-4
+        width_star = .1
+        if np.nanmean(folded.flux[t < 0]) > np.nanmean(folded.flux[t > 0]):
+            to_star = .25
+        else:
+            to_star = -.25
+        for i in range(niters):
```
Comment (barentsen) [inline lightkurve/lightcurve.py]: Do you want to add a few lines of comments inside these for loops, just to guide the untrained eye where in the algorithm we are?

**Comment by barentsen** [comment]
Thanks!  LGTM :+1:

**Pull request merged**